### PR TITLE
Add reno recipe

### PR DESCRIPTION
**Pull request by wshanks**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by wshanks** [comment]
@conda-forge/help-python ready for review

**Comment by wshanks** [comment]
This is grayskull output with no modifications.

File: recipes/reno/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/reno-{{ version }}.tar.gz
+  sha256: 0ebcad207133f4dbb7bc176ec3513fcd8e107a5d9e2f04b5621eb61da1147089
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
```
Comment (ocefpaf) [inline recipes/reno/meta.yaml:15]: This should fix the Windows build.

```suggestion
  number: 0
  entry_points:
    - reno = reno.main:main
```

File: recipes/reno/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    - pyyaml >=5.3.1
+    - dulwich >=0.15.0
+    - packaging >=20.4
+    - pbr
+
+test:
+  imports:
+    - reno
+  commands:
+    - pip check
```
Comment (wshanks) [inline recipes/reno/meta.yaml:35]:
```suggestion
    - pip check
    - reno --help
```

**Comment by wshanks** [comment]
Thanks! I should have waited for the CI before requesting review.... I am wondering if I should include the optional sphinx extra dependencies. Personally I use reno without specifically `[sphinx]` with sphinx (just install it separately any way because it's needed to build the docs), so I don't think it's that important.

**Pull request merged**